### ISSUE TITLE
Table column reordering slow performance

### ISSUE THREAD

**Issue by alinvdumitru** [schrodinger/fixed-data-table-2#138]
**Table Column Reordering Dragging is slow under 15-16 columns.**

## Expected Behavior
Table columns header should be easy to drag around.

## Current Behavior
We experience some slow dragging movement when we have 15-16 columns in view with lot of data. We looked and introduced ```allowCellsRecycling``` which increases performance slightly (because we have columns that are out of the view), but not a notable improvement. The performance is even worse for 20 columns (in viewport) and makes reordering really hard if page is heavy on data and updated often.

- I made an isolation test on your example for dragging and it does the same for 15-16 columns.
- Looking at FixedTableDataCell it looks like cells are re-rendering every time a column is dragged.

## Possible Solution
- A suggestion would be to introduce a performance check on `shouldComponentUpdate` which should only return true for ```headers``` or  ```scrolling``` out of view when dragging since table itself only changes structure on reordering end.

- Right now it seems ```shouldComponentUpdate``` always returns true on dragging.

* Version used: 0.7.11
* Browser Name and version: Chrome/Mozilla
* Operating System and version (desktop or mobile): Windows

**Comment by KamranAsif** [comment]
Hey,

We very recently introduced a flag called [pureRendering](http://schrodinger.github.io/fixed-data-table-2/api-column.html#-purerendering-). Can you try adding that to your columns to see if it helps?

**Comment by wcjordan** [comment]
Also if you could give us an idea of the changes you made to reproduce this on the examples page, that would be helpful as well.

**Comment by alinvdumitru** [comment]
Hey @KamranAsif @wcjordan. 

Indeed ```pureRendering``` seems what I was looking for. Although after introducing it the performance didn't improve (or at least I didn't notice any difference) even though only the headers were re-rendering now.

The only change I did to reorder example was to add more columns and fit them into view (we usually have 15-16 or even more). After adding ```pureRendering``` to the example the performance improved notably.

We'll investigate further optimization but regarding fixed-data-table ```pureRendering``` was what we aimed for.
